# 02-05-2019

# Expression analysis

First I will merge all the `.bam` files generated into one file for each condition. In order to do so I will use `samtools merge`. Since the bam files are already sorted, this shouldn't take much time. 

Okay, so I am not going to merge the reads because they are biological replicates and I want to keep them separate. A reason for that is the posterior statistical analysis. 

So now I can identify transcriptional units. For that I can use IGV and load both the`.gtf` annotation file and the `.bam` file. 

Meanwhile I can do the synteny analysis. For that I will use the following: 

```bash
module load bioinfo-tools
module load blast
makeblastdb -in stunning-garbanzo/results/genomics/pilon_assembly/Pilon_assembly_05-04-2019.fasta -dbtype nucl
blastn -query Genome2 -db Genome1 -evalue 1 -task megablast -outfmt 6 > Genome1_Genome2.crunch
```

Then I will use ACT tool from Artemis in order to analyse the synteny. For that I will need the annotation file from my assembly as well as the fasta, and the same for the paper annotation and a closely relared species annotation. Then I will need the `.crunch` file which contains the blast comparisson coordinates. 
 
To merge all the `.gbk` files from the paper assembly I used a `python` package called (merge-gbk-records)[https://github.com/kblin/merge-gbk-records]

```python
merge-gbk-records results/genomics/paper_annotation/*.gbk > merged.gbk
```



