# 04-04-2019

```console
ssh -AX clno2215@rackham.uppmax.uu.se

```

Connected to Uppmax

Today's plan: get the bash script for the genome assembly + getting to know the data 


The PacBio filenames are the following: 

-m131023_233432_42174_c100519312550000001823081209281335_s1_X0.1.subreads.fastq.gz
-m131023_233432_42174_c100519312550000001823081209281335_s1_X0.1.subreads.fastq.gz.save
-m131023_233432_42174_c100519312550000001823081209281335_s1_X0.2.subreads.fastq.gz
-m131023_233432_42174_c100519312550000001823081209281335_s1_X0.3.subreads.fastq.gz
-m131024_200535_42174_c100563672550000001823084212221342_s1_p0.1.subreads.fastq.gz
-m131024_200535_42174_c100563672550000001823084212221342_s1_p0.2.subreads.fastq.gz
-m131024_200535_42174_c100563672550000001823084212221342_s1_p0.3.subreads.fastq.gz

My softlinks are the following
We don't have the data from the reads because it's not publicly available. 
We don't need to `cat` the data into one single fasta file. Instead we run Canu on the whole folder with the reads. 


```console
canu -p canu.04-04-2019 -d canu_outdir.04-04-2019 genomeSize=3.1m -pacbio-raw /home/clno2215/stunning-garbanzo/data/genomics/raw_data/*_PacBio_* 
```

The corresponding batch script:

```console
#!/bin/bash -l
#SBATCH -A g2019003
#SBATCH -p core
#SBATCH -n 2
#SBATCH -t 06:00:00
#SBATCH -J canu-04-04-2019
#SBATCH --mail-type=ALL
#SBATCH --mail-user c.novella.rausell@gmail.com
# Load modules
module load bioinfo-tools
module load canu
# Your commands
canu -p canu.04-04-2019 -d canu_outdir.04-04-2019 genomeSize=3.1m -pacbio-raw /home/clno2215/stunning-garbanzo/data/genomics/raw_data/*_PacBio_* 
```
