## 05-04-2019

Today we are going to take a look at the assembly we got running yesterday. The results are in `canu_outdir.04-04-2019`. The files are the following:

-canu.04-04-2019.contigs.fasta
-canu.04-04-2019.contigs.gfa
-canu.04-04-2019.contigs.layout
-canu.04-04-2019.contigs.layout.readToTig
-canu.04-04-2019.contigs.layout.tigInfo
-canu.04-04-2019.correctedReads.fasta.gz
-canu.04-04-2019.gkpStore/
-canu.04-04-2019.gkpStore.err
-canu.04-04-2019.gkpStore.gkp
-canu.04-04-2019.report
-canu.04-04-2019.trimmedReads.fasta.gz
-canu.04-04-2019.unassembled.fasta
-canu.04-04-2019.unitigs.bed
-canu.04-04-2019.unitigs.fasta
-canu.04-04-2019.unitigs.gfa
-canu.04-04-2019.unitigs.layout
-canu.04-04-2019.unitigs.layout.readToTig
-canu.04-04-2019.unitigs.layout.tigInfo
-canu-logs/
-canu-scripts/
-correction/
-trimming/
-unitigging/

```bash
head canu_outdir.04-04-2019.contigs.layout.tigInfo
## out
tigID	tigLen	coordType	covStat	coverage	tigClass	sugReptsugCirc	numChildren
1	2773595	ungapped	3547.40	36.21	contig	no	yes	12040
19	40016	ungapped	47.43	28.58	contig	no	yes	143
20	44600	ungapped	117.06	10.52	contig	no	no	64
25	12415	ungapped	13.89	4.50	contig	no	no	7
27	26279	ungapped	63.11	5.99	contig	no	no	19
29	11988	ungapped	18.16	6.46	contig	no	no	11
86	7860	ungapped	0.00	1.00	unassm	no	no	1
87	7338	ungapped	0.00	1.00	unassm	no	no	1
88	7628	ungapped	0.00	1.00	unassm	no	no	1

```
I'm going to change the tab generated table to comma separated so I can include it on the report. For that I'm going to use `tr`.

```bash
head canu.04-04-2019.contigs.fasta >/home/clno2215/stunning-garbanzo/results/genomics/canu_contigs

[../results/genomics]$ cat canu_contigs.tsv | tr "\\t" ","  > canu_contigs.csv
```

|#tigID|tigLen                       |coordType|covStat                                      |coverage|tigClass|sugRept|sugCirc|numChildren|
|------|-----------------------------|---------|---------------------------------------------|--------|--------|-------|-------|-----------|
|1     |2773595                      |ungapped |3547.40                                      |36.21   |contig  |no     |yes    |12040      |
|19    |40016                        |ungapped |47.43                                        |28.58   |contig  |no     |yes    |143        |
|20    |44600                        |ungapped |117.06                                       |10.52   |contig  |no     |no     |64         |
|25    |12415                        |ungapped |13.89                                        |4.50    |contig  |no     |no     |7          |
|27    |26279                        |ungapped |63.11                                        |5.99    |contig  |no     |no     |19         |
|29    |11988                        |ungapped |18.16                                        |6.46    |contig  |no     |no     |11         |
|86    |7860                         |ungapped |0.00                                         |1.00    |unassm  |no     |no     |1          |
|87    |7338                         |ungapped |0.00                                         |1.00    |unassm  |no     |no     |1          |
|88    |7628                         |ungapped |0.00                                         |1.00    |unassm  |no     |no     |1          |

As observed we got the main chromosome with high coverage (being this contig #tigID 1) with a total length of 2773595 bp. It is already circularized so my initial idea of running `circlator` is discarded at this moment. The coverage seems good enough to keep the chromosome but we have a bunch of contigs with low coverage.

According to my plan now we  should correct this assembly with the Illumina reads that were quality checked before (see `04-04-2019.md`). The quality of the reads was [high enough]( http://htmlpreview.github.com/?https://github.com/Slakky/stunning-garbanzo/blob/master/results/genomics/fastqc/01_Illumina_1_clean_fastqc.html) to not trim them.

In  order to correct we align the `Illumina`reads to the assembly. For that we use `bwa-mem`.

```bash
module load bioinfo-tools
module load bwa
module load samtools/1.9

bwa index /home/clno2215/canu_outdir.04-04-2019/canu.04-04-2019.contigs.fasta

[bwa_outdir.05-04-2019]bwa mem -M -p bwa -t 2 /home/clno2215/canu_outdir.04-04-2019/canu.04-04-2019.contigs.fasta /home/clno2215/stunning-garbanzo/data/genomics/raw_data/01_Illumina_1_clean_fastqc.gz /home/clno2215/stunning-garbanzo/data/genomics/raw_data/02_Illumina_2_clean_fastqc.gz | samtools sort > aln.05-04-2019.bam
```
The corresponding batch scripts
```bash
#!/bin/bash -l
#SBATCH -A g2019003
#SBATCH -p core
#SBATCH -n 2
#SBATCH -t 05:00:00
#SBATCH -J bwa-05-04-2019
#SBATCH --mail-type=ALL
#SBATCH --mail-user c.novella.rausell@gmail.com
# Load modules
module load bioinfo-tools
module load bwa
module load samtools/1.9
# Your commands
bwa index /home/clno2215/canu_outdir.04-04-2019/canu.04-04-2019.contigs.fasta
cd /home/clno2215/bwa_outdir.05-04-2019
bwa mem -M -p bwa -t 2 /home/clno2215/canu_outdir.04-04-2019/canu.04-04-2019.contigs.fasta /home/clno2215/stunning-garbanzo/data/genomics/raw_data/01_Illumina_1_clean_fastqc.gz /home/clno2215/stunning-garbanzo/data/genomics/raw_data/02_Illumina_2_clean_fastqc.gz > aln.05-04-2019.sam

```

The output format is `.sam` which is not compiled and weight a lot

```bash
samtools view -Sb aln.05-04-2019.sam > aln.05-04-2019.bam
samtools sort aln.05-04-2019.bam
```
Now we have `aln.05-04-2019.bam`.

We have to index it using `samtools`:

```bash
samtools index aln.05-04-2019.bam
```

Once we've done that. We run Pilon to correct assembly errors with our alignment file and the assembly files

```bash
module load Pilon/1.22
pilon --genome /home/clno2215/canu_outdir.04-04-2019/canu.04-04-2019.contigs.fasta --frags /home/clno2215/bwa_outdir.05-04-2019/aln.05-04-2019.bam --outoput pilon_outdir.05-04-2019 --fix all --threads 2 --changes --verbose

```
