# 07-05-2019

I had to run `samtools` sort and index on the `.bam` files.

This is the batch script I ran:

```bash
#!/bin/bash -l
#SBATCH -A g2019003
#SBATCH -p core
#SBATCH -n 2
#SBATCH -t 04:00:00
#SBATCH -J SamTools_07-05-2019
#SBATCH --mail-type=ALL
#SBATCH --mail-user c.novella.rausell@gmail.com
# Load modules
module load bioinfo-tools
module load samtools/1.9
# Your commands
cd /home/clno2215/stunning-garbanzo/results/transcriptomics/BH/bowtie2
samtools sort -o efaec_pilon_bwt2_30-04-2019_ERR1797973_sorted.bam efaec_pilon_bwt2_30-04-2019_ERR1797973.bam
samtools index efaec_pilon_bwt2_30-04-2019_ERR1797973_sorted.bam
rm efaec_pilon_bwt2_30-04-2019_ERR1797973.bam
samtoools sort -o efaec_pilon_bwt2_30-04-2019_ERR1797974_sorted.bam  efaec_pilon_bwt2_30-04-2019_ERR1797974.bam
samtools index efaec_pilon_bwt2_30-04-2019_ERR1797974_sorted.bam
rm efaec_pilon_bwt2_30-04-2019_ERR1797974.bam
cd /home/clno2215/stunning-garbanzo/results/transcriptomics/Serum/bowtie2
samtools sort -o efaec_pilon_bwt2_30-04-2019_ERR1797969_sorted.bam efaec_pilon_bwt2_30-04-2019_ERR1797969.bam
samtools index efaec_pilon_bwt2_30-04-2019_ERR1797969_sorted.bam
rm efaec_pilon_bwt2_30-04-2019_ERR1797969.bam
samtoools sort -o efaec_pilon_bwt2_30-04-2019_ERR1797970_sorted.bam  efaec_pilon_bwt2_30-04-2019_ERR1797970.bam
samtools index efaec_pilon_bwt2_30-04-2019_ERR1797970_sorted.bam
rm efaec_pilon_bwt2_30-04-2019_ERR1797970.bam
samtoools sort -o efaec_pilon_bwt2_30-04-2019_ERR1797971_sorted.bam  efaec_pilon_bwt2_30-04-2019_ERR1797971.bam
samtools index efaec_pilon_bwt2_30-04-2019_ERR1797971_sorted.bam
rm efaec_pilon_bwt2_30-04-2019_ERR1797971.ba
```
I am going to use `HTseq` to do the read counts. It needs both a bam and an annotation file. Apparently he doesn't like FASTA sequences in the `gff` file, so we can do the following to remove the FASTA sequences:

```bash
grep -n FASTA <file.gff>
head -n <line-1> <file.gff> > <output.gff>
```


The bash script to run `HTSeq` with the `.gtf` file produced on 29-05-2019 with a python script:

```bash
#!/bin/bash -l
#SBATCH -A g2019003
#SBATCH -p core
#SBATCH -n 2
#SBATCH -t 04:00:00
#SBATCH -J HTseq-count_07-05-2019
#SBATCH --mail-type=ALL
#SBATCH --mail-user c.novella.rausell@gmail.com
# Load modules
module load bioinfo-tools
module load htseq/0.9.1
# Your commands
cd /home/clno2215/stunning-garbanzo/results/transcriptomics/BH/bowtie2
for i in *_sorted.bam
do
echo $i
htseq-count $i /home/clno2215/stunning-garbanzo/results/genomics/prokka_annotation/PROKKA_04292019.gtf -f bam -t CDS > $i.count
done
cd /home/clno2215/stunning-garbanzo/results/transcriptomics/Serum/bowtie2
for i in *_sorted.bam
do
echo $i
htseq-count $i /home/clno2215/stunning-garbanzo/results/genomics/prokka_annotation/PROKKA_04292019.gtf -f bam -t CDS > $i.count
done
```
>>> Add the Synteny analysis done with ACT and the annotated chromosome vs chromosome of the paper!
