# 11-04-2019

We have the corrected genome assembly by `Pilon`. It is on the `pilonoutdirectory`. 


## Quality assesment of the assembly

For this I'm going to get the following parameters:

-QUAST run
    -This will give use some statistics about the assembly, such as:
        -N50
        -L50
        -Length of contigs
        ...


### QUAST

`quast.py pilon_outdir.05-04-2019/pilon_outdir.sorted.05-04-2019.fasta -o Canu_Pilon_Quast`

`quast.py pilon_outdir.05-04-2019/pilon_outdir.sorted.05-04-2019.fasta -R reference -o Canu_Pilon_Quast_Reference`
>Reference used: [Enterococcus faecium](https://www.ncbi.nlm.nih.gov/genome/?term=Enterococcus+faecium+DO).

`quast.py spades_outdir.04-04-2019/scaffolds.fasta -o SPAdes_Quast`

`quast.py spades_outdir.04-04-2019/scaffolds.fasta -R reference -o SPAdes_Quast_Reference`
>Reference used: [Enterococcus faecium](https://www.ncbi.nlm.nih.gov/genome/?term=Enterococcus+faecium+DO)

I changed my mind, I'm going to use the assembled genome from the paper as a reference:

In order to download the contigs from NCBI:

```bash
wget -O <accession> 'https://www.ncbi.nlm.nih.gov/search/api/sequence/<accession>/?report=fasta'
```
In order to put all the contigs into one reference file:

```bash
cat CP*.fasta > Assembled_genome.fasta
```

Getting rid of newlines:

```bash
awk '!/^>/ { printf "%s", $0; n = "\n" } 
/^>/ { print n $0; n = "" }
END { printf "%s", n }
' Assembled_genome.fasta
```


```bash
quast.py pilon_outdir.05-04-2019/pilon_outdir.sorted.05-04-2019.fasta -o stunning-garbanzo/results/genomics/quast/Canu_Pilon_Quast

quast.py pilon_outdir.05-04-2019/pilon_outdir.sorted.05-04-2019.fasta -R stunning-garbanzo/results/genomics/paper_assembly/* -o stunning-garbanzo/results/genomics/quast/Canu_Pilon_Quast_Reference


quast.py spades_outdir.04-04-2019/scaffolds.fasta -o stunning-garbanzo/results/genomics/quast/SPAdes_Quast

quast.py spades_outdir.04-04-2019/scaffolds.fasta -R stunning-garbanzo/results/genomics/paper_assembly/* -o stunning-garbanzo/results/genomics/quast/SPAdes_Quast_Reference

```

Now we take a look at the results:

Report from `Canu_Pilon_Quast`

|                            |                                | 
|----------------------------|--------------------------------| 
| Assembly                   | pilon_outdir.sorted.05-04-2019 | 
| # contigs (>= 0 bp)        | 8                              | 
| # contigs (>= 1000 bp)     | 8                              | 
| # contigs (>= 5000 bp)     | 8                              | 
| # contigs (>= 10000 bp)    | 7                              | 
| # contigs (>= 25000 bp)    | 5                              | 
| # contigs (>= 50000 bp)    | 2                              | 
| Total length (>= 0 bp)     | 3136314                        | 
| Total length (>= 1000 bp)  | 3136314                        | 
| Total length (>= 5000 bp)  | 3136314                        | 
| Total length (>= 10000 bp) | 3128014                        | 
| Total length (>= 25000 bp) | 3103592                        | 
| Total length (>= 50000 bp) | 2992679                        | 
| # contigs                  | 8                              | 
| Largest contig             | 2773702                        | 
| Total length               | 3136314                        | 
| GC (%)                     | 37.79                          | 
| N50                        | 2773702                        | 
| N75                        | 2773702                        | 
| L50                        | 1                              | 
| L75                        | 1                              | 
| # N's per 100 kbp          | 0.00                           | 

Report from `Canu_Pilon_Quast_Reference`

|                             |                                | 
|-----------------------------|--------------------------------| 
| Assembly                    | pilon_outdir.sorted.05-04-2019 | 
| # contigs (>= 0 bp)         | 8                              | 
| # contigs (>= 1000 bp)      | 8                              | 
| # contigs (>= 5000 bp)      | 8                              | 
| # contigs (>= 10000 bp)     | 7                              | 
| # contigs (>= 25000 bp)     | 5                              | 
| # contigs (>= 50000 bp)     | 2                              | 
| Total length (>= 0 bp)      | 3136314                        | 
| Total length (>= 1000 bp)   | 3136314                        | 
| Total length (>= 5000 bp)   | 3136314                        | 
| Total length (>= 10000 bp)  | 3128014                        | 
| Total length (>= 25000 bp)  | 3103592                        | 
| Total length (>= 50000 bp)  | 2992679                        | 
| # contigs                   | 8                              | 
| Largest contig              | 2773702                        | 
| Total length                | 3136314                        | 
| Reference length            | 3168410                        | 
| GC (%)                      | 37.79                          | 
| Reference GC (%)            | 37.70                          | 
| N50                         | 2773702                        | 
| NG50                        | 2773702                        | 
| N75                         | 2773702                        | 
| NG75                        | 2773702                        | 
| L50                         | 1                              | 
| LG50                        | 1                              | 
| L75                         | 1                              | 
| LG75                        | 1                              | 
| # misassemblies             | 7                              | 
| # misassembled contigs      | 5                              | 
| Misassembled contigs length | 3052985                        | 
| # local misassemblies       | 0                              | 
| # unaligned mis. contigs    | 0                              | 
| # unaligned contigs         | 0 + 0 part                     | 
| Unaligned length            | 0                              | 
| Genome fraction (%)         | 98.119                         | 
| Duplication ratio           | 1.009                          | 
| # N's per 100 kbp           | 0.00                           | 
| # mismatches per 100 kbp    | 3.89                           | 
| # indels per 100 kbp        | 4.21                           | 
| Largest alignment           | 2195416                        | 
| Total aligned length        | 3136310                        | 
| NA50                        | 2195416                        | 
| NGA50                       | 2195416                        | 
| NA75                        | 578286                         | 
| NGA75                       | 578286                         | 
| LA50                        | 1                              | 
| LGA50                       | 1                              | 
| LA75                        | 2                              | 
| LGA75                       | 2                              | 


I'm going to run `SPAdes` again, because I got too many contigs. Removing some of the options I added to the command, right now the script looks like this: 

```bash
module load bioinfo-tools
module load spades
spades.py -t 2 --nanopore /home/clno2215/stunning-garbanzo/data/genomics/raw_data/03_Nanopore_all.fasta.gz  -1 /home/clno2215/stunning-garbanzo/data/genomics/raw_data/01_Illumina_1_clean.fq.gz -2 /home/clno2215/stunning-garbanzo/data/genomics/raw_data/02_Illumina_2_clean.fq.gz -o spades_outidr.11-04-2019
```

In order to identify the plasmids with the contigs that we got out of the assembly i'm going to split the whole assembly into different contigs. For that I used the following command:

```bash
split -l 2 Assembly_genome.fasta
```
Following this step we can do a `mmer plot` (basically a BLAST) with the contig that aligns to the corresponding contig in the reference. We can fetch this information in the output file `/contig_reports/all_alignment.tsv`. 

But that's something for tomorrow. Before leaving I ran Prokka with the assembly out of canu `Pilon_assembly` folder. For that the following command was submited as a _sbatch_ job:

```bash
prokka /home/clno2215/stunning-garbanzo/results/genomics/pilon_assembly/Pilon_assembly_05-04-2019.fasta --outdir /home/clno2215/stunning-garbanzo/results/genomics/prokka_annotation/
```
