# 13-05-2019

Today I'm going to proceed with the differential gene expression. For that I pulled my count files from my github repositoy (they are not so big). I'm going to analyze them on my laptop using `R`.

I have renamed the count files to "0(replicate number) _ Condition _ counts.tab".

The htseq files contain some statistics that are not relevant for the differential expression analysis, because of that I'm going to remove that information using the following script (also in the `code` folder):

```bash
for filename in *.tab; do
    head -3044 "$filename" > "output_$filename"
done
```
```python
files = list.files(path=".")
file1 = read.table(files[1], col.names=c("gene_id",files[1]))
file2 = read.table(files[2], col.names=c("gene_id",files[2]))
out.file = merge (file1, file2, by=c("gene_id"))
for(i in 3:length(files))
{
  file = read.table(files[i],col.names=c("gene_id",files[i]))
  out.file <- merge(out.file, file, by=c("gene_id"))
}

write.table(out.file, file = "htseq_all_sample_count.tsv",sep="\t", row.names = FALSE)
```
After I have the file `htseq_all_sample_counts.tsv` I can use that as an input for `DESeq2`. The table contains all the genes counts for every replicate and condition in the experiment (BH and Serum).

The following steps are performed in the RMarkdown notebook

