# 16 - 05 - 2019

Today I'm going to start working on the extra analyses. If I remember correctly, the assembly from Canu was not properly assembled since it had some gaps on some contigs. I'm going to use the Spades assembly in order to properly assembly those parts. 

One approach that comes into mind is to look at the gaps and blast the sorrounding sequences into the SPAdes assembly. 

|                                                                            |                   |               | 
|----------------------------------------------------------------------------|-------------------|---------------| 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | tig00000001_pilon | Contig number | 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | tig00000001_pilon | Contig1       | 
| CP014531.1_Enterococcus_faecium_strain_E745_plasmid_pl2__complete_sequence | tig00000019_pilon | Contig2       | 
| CP014531.1_Enterococcus_faecium_strain_E745_plasmid_pl2__complete_sequence | tig00000019_pilon | Contig2       | 
| CP014534.1_Enterococcus_faecium_strain_E745_plasmid_pl5__complete_sequence | tig00000020_pilon | Contig3       | 
| CP014533.1_Enterococcus_faecium_strain_E745_plasmid_pl4__complete_sequence | tig00000025_pilon | Contig4       | 
| CP014535.1_Enterococcus_faecium_strain_E745_plasmid_pl6__complete_sequence | tig00000027_pilon | Contig5       | 
| CP014532.1_Enterococcus_faecium_strain_E745_plasmid_pl3__complete_sequence | tig00000029_pilon | Contig6       | 
| CP014532.1_Enterococcus_faecium_strain_E745_plasmid_pl3__complete_sequence | tig00000029_pilon | Contig6       | 
| CP014532.1_Enterococcus_faecium_strain_E745_plasmid_pl3__complete_sequence | tig00000029_pilon | Contig6       | 
| CP014530.1_Enterococcus_faecium_strain_E745_plasmid_pl1__complete_sequence | tig00000464_pilon | Contig7       | 
| CP014534.1_Enterococcus_faecium_strain_E745_plasmid_pl5__complete_sequence | tig00000464_pilon | Contig7       | 
| CP014530.1_Enterococcus_faecium_strain_E745_plasmid_pl1__complete_sequence | tig00000464_pilon | Contig7       | 
| CP014530.1_Enterococcus_faecium_strain_E745_plasmid_pl1__complete_sequence | tig00000465_pilon | Contig8       | 
| CP014530.1_Enterococcus_faecium_strain_E745_plasmid_pl1__complete_sequence | tig00000465_pilon | Contig8       | 

The missasembled contigs report from `quast`:
tig00000001_pilon
Extensive misassembly (relocation, inconsistency = -8697 [linear representation of circular genome]) between 578286 1 and 2773702 578287
tig00000019_pilon
Extensive misassembly (relocation, inconsistency = -7592 [linear representation of circular genome]) between 13290 1 and 40017 13291
tig00000029_pilon
Extensive misassembly (relocation, inconsistency = -1341 [linear representation of circular genome]) between 1 1346 and 1347 10663
Extensive misassembly (relocation, inconsistency = -1325 [linear representation of circular genome]) between 1347 10663 and 10664 11989
tig00000464_pilon
Extensive misassembly (translocation) between 183449 1 and 202668 183450
Extensive misassembly (translocation) between 191799 183450 and 218977 191800
tig00000465_pilon
Extensive misassembly (relocation, inconsistency = 1441) between 6563 1 and 8300 6564

And in the SPAdes assembly
|                                                                            |                                     |           | 
|----------------------------------------------------------------------------|-------------------------------------|-----------| 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | NODE_1_length_986763_cov_43.494015  | Contig_00 | 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | NODE_2_length_840748_cov_42.698417  | Contig_01 | 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | NODE_3_length_500609_cov_47.960903  | Contig_02 | 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | NODE_3_length_500609_cov_47.960903  | Contig_02 | 
| CP014530.1_Enterococcus_faecium_strain_E745_plasmid_pl1__complete_sequence | NODE_4_length_266941_cov_102.662627 | Contig_03 | 
| CP014534.1_Enterococcus_faecium_strain_E745_plasmid_pl5__complete_sequence | NODE_4_length_266941_cov_102.662627 | Contig_03 | 
| CP014530.1_Enterococcus_faecium_strain_E745_plasmid_pl1__complete_sequence | NODE_4_length_266941_cov_102.662627 | Contig_03 | 
| CP014534.1_Enterococcus_faecium_strain_E745_plasmid_pl5__complete_sequence | NODE_4_length_266941_cov_102.662627 | Contig_03 | 
| CP014530.1_Enterococcus_faecium_strain_E745_plasmid_pl1__complete_sequence | NODE_4_length_266941_cov_102.662627 | Contig_03 | 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | NODE_5_length_164595_cov_65.919916  | Contig_04 | 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | NODE_6_length_150922_cov_42.321628  | Contig_05 | 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | NODE_7_length_88799_cov_43.513274   | Contig_06 | 
| CP014535.1_Enterococcus_faecium_strain_E745_plasmid_pl6__complete_sequence | NODE_8_length_62897_cov_74.024920   | Contig_07 | 
| CP014535.1_Enterococcus_faecium_strain_E745_plasmid_pl6__complete_sequence | NODE_8_length_62897_cov_74.024920   | Contig_07 | 
| CP014531.1_Enterococcus_faecium_strain_E745_plasmid_pl2__complete_sequence | NODE_9_length_31334_cov_224.503597  | Contig_08 | 
| CP014531.1_Enterococcus_faecium_strain_E745_plasmid_pl2__complete_sequence | NODE_9_length_31334_cov_224.503597  | Contig_08 | 
| CP014533.1_Enterococcus_faecium_strain_E745_plasmid_pl4__complete_sequence | NODE_10_length_12383_cov_107.950438 | Contig_09 | 
| CP014533.1_Enterococcus_faecium_strain_E745_plasmid_pl4__complete_sequence | NODE_10_length_12383_cov_107.950438 | Contig_09 | 
| CP014532.1_Enterococcus_faecium_strain_E745_plasmid_pl3__complete_sequence | NODE_11_length_9325_cov_146.068608  | Contig_10 | 
| CP014532.1_Enterococcus_faecium_strain_E745_plasmid_pl3__complete_sequence | NODE_11_length_9325_cov_146.068608  | Contig_10 | 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | NODE_12_length_5232_cov_70.113772   | Contig_11 | 
| CP014534.1_Enterococcus_faecium_strain_E745_plasmid_pl5__complete_sequence | NODE_13_length_5228_cov_42.845544   | Contig_12 | 
| CP014531.1_Enterococcus_faecium_strain_E745_plasmid_pl2__complete_sequence | NODE_14_length_3460_cov_267.035242  | Contig_13 | 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | NODE_16_length_1429_cov_27.793304   | Contig_14 | 
| CP014531.1_Enterococcus_faecium_strain_E745_plasmid_pl2__complete_sequence | NODE_17_length_886_cov_350.894103   | Contig_15 | 
| CP014529.1_Enterococcus_faecium_strain_E745__complete_genome               | NODE_18_length_519_cov_32.366379    | Contig_16 | 


The misasssembled_contig report out of `quast`:
NODE_4_length_266941_cov_102.662627
Extensive misassembly (translocation) between 201456 1 and 211048 201457
Extensive misassembly (translocation) between 206632 201457 and 221487 206633
Extensive misassembly (translocation) between 215405 206633 and 255501 215406
Extensive misassembly (translocation) between 255501 215406 and 266941 255502
NODE_10_length_12383_cov_107.950438
Extensive misassembly (relocation, inconsistency = -10556) between 1 10198 and 10199 12383

On the Canu assembly we have only one contig that aligns to the 6th plasmid, but in. the SPades we have 2 contigs. I'm going to blast the contig from Canu to the contigs of SPAdes, to know if I can replace the Canu contig with the better assembled Canu assembly.

The blast output shows that there is a gap in the alignment, but I used the SPAdes as a subject and the Canu as quuery. I substitued the Contig5 with the NODE8. I have a new folder with the improved assembly `pilon_assembly_improved_SPAdes`

