# 25 - 04 - 2019

# Transcriptome assembly

Today I'm going to start to work with the RNA data. We have both paired and single reads in two conditions:

- Serum
- BH

The ultimate goal is to compare the gene expression in this two conditions using both the assembled genome and the RNA reads.

The authors of the paper used the paired end Illumina reads to map them to the genome and obtain the transcriptome assembly. 

But firs we have to check the quality of the reads and trim them if necessary. 

For that we run `FastQC`. 

 ```bash
 fastqc Serum/*_paired_* -o /home/clno2215/stunning-garbanzo/results/transcriptomics/Serum/fastqc/

 fastqc BH/*_paired_* -o /home/clno2215/stunning-garbanzo/results/transcriptomics/BH/fastqc/
 ```

The result is on the `results` directory. Even though most reads seem to have pretty good quality I'm going to trim them anyway, just to make sure there are no other artifacts.

After that we can run `trimmomatic` on the files to check if they are really trimmed. For that I ran the following sbatch script:

```bash
#!/bin/bash -l
#SBATCH -A g2019003
#SBATCH -p core
#SBATCH -n 2
#SBATCH -t 05:00:00
#SBATCH -J trim_serum_25-04-2019
#SBATCH --mail-type=ALL
#SBATCH --mail-user c.novella.rausell@gmail.com
# Load modules
module load bioinfo-tools
module load trimmomatic
# Your commands
cd /home/clno2215/stunning-garbanzo/data/transcriptomics/RNA-seq/raw_data/Serum
bash cmd_file
```
`cmd_file` contain the trimmomatic commands needed for every file. To generate these files I ran `09_BashScript_Trim.sh` on the folder containing the reads. I manually modified some commands since the script was not working as intended. 

Once `trimmomatic` has finished, we can check if the quality of the reads has significantly increased compared to the previous available. For that we run `FastQC` again. 

```bash
#!/bin/bash -l
#SBATCH -A g2019003
#SBATCH -p core
#SBATCH -n 2
#SBATCH -t 05:00:00
#SBATCH -J fastqc_trimmed_29-04-2019
#SBATCH --mail-type=ALL
#SBATCH --mail-user c.novella.rausell@gmail.com
# Load modules
module load bioinfo-tools
module load fastqc
# Your commands
fastqc /home/clno2215/stunning-garbanzo/data/transcriptomics/RNA-seq/trimmed_data/BH/*_paired.fq* -o /home/clno2215/stunning-garbanzo/results/transcriptomics/Serum/fastqc_trimmed/

fastqc /home/clno2215/stunning-garbanzo/data/transcriptomics/RNA-seq/trimmed_data/Serum/*_paired.fq* -o /home/clno2215/stunning-garbanzo/results/transcriptomics/BH/fastqc_trimmed/
```

The results are the expected considering that we trimmed what was given as already trimmed. I'm going to include a couple of screenshots of the report given by FastQC. 

<p align="center">
    <img src="../../images_github/fastqc1_avgq.png" width="450"/>
    <img src="../../images_github/fastqc1_adptr.png" width="450"/>
    <img src="../../images_github/fastqc2_avgq.png" width="450"/>
    <img src="../../images_github/fastqc2_adptr.png" width="450"/>
</p>
